src/Server/MySQLHandler.cpp

```diff
@@ -222,7 +222,7 @@ void MySQLHandler::authenticate(const String & user_name, const String & auth_pl
         // For compatibility with JavaScript MySQL client, Native41 authentication plugin is used when possible (if password is specified using double SHA1). Otherwise SHA256 plugin is used.
         auto user = connection_context.getAccessControlManager().read<User>(user_name);
         const DB::Authentication::Type user_auth_type = user->authentication.getType();
-        if (user_auth_type != DB::Authentication::DOUBLE_SHA1_PASSWORD && user_auth_type != DB::Authentication::PLAINTEXT_PASSWORD && user_auth_type != DB::Authentication::NO_PASSWORD)
+        if (user_auth_type != DB::Authentication::LDAP_SERVER && user_auth_type != DB::Authentication::DOUBLE_SHA1_PASSWORD && user_auth_type != DB::Authentication::PLAINTEXT_PASSWORD && user_auth_type != DB::Authentication::NO_PASSWORD)
```

Comment:
It looks like it should be simplified.

Reply:
Well, I was just following the pattern there. Let me know, what kind of simplification you have in mind.

Reply:
Just invert the condition:

```
if (user_auth_type != DB::Authentication::LDAP_SERVER && user_auth_type != DB::Authentication::DOUBLE_SHA1_PASSWORD && user_auth_type != DB::Authentication::PLAINTEXT_PASSWORD && user_auth_type != DB::Authentication::NO_PASSWORD)
```

=>

```
if (user_auth_type == DB::Authentication::SHA256_PASSWORD)
```

Reply:
OK, will do. I wonder why it wasn't set to it before my changes though...

Reply:
Done.